### DISCUSSION TITLE
How to add a custom attribute

### DISCUSSION THREAD

**Discussion by dvv** [kanidm/kanidm#2612] [Q&A]
Hi!

So far I over-abuse `legalname` for an opaque json but I do want to leave this truly private attribute to the person's control.
I drafted the diff here https://github.com/dvv/kanidm/commit/60285f07a3e92c2e124a64cb1bb30aba9d29cf97 but it's completely by naive try-error.
Please validate/comment on the procedure of a custom attribute addition.

TIA

**Comment by Firstyear** [comment]
We have a longer road before we will support custom schema. See https://github.com/kanidm/kanidm/issues/252

**Reply by dvv** [reply]
I understand. But I need to explore that in a shorter way. Please comment what should I take care of in addition to https://github.com/dvv/kanidm/commit/60285f07a3e92c2e124a64cb1bb30aba9d29cf97 ?

**Reply by yaleman** [reply]
Sorry, but we have incredibly limited resources. If it was a PR directly implementing a feature that's part of an accepted design we could help, but not an unsupportable branch of the code.

**Reply by dvv** [reply]
That's exactly why I'm asking for initial assistance -- to offload the further burden. If I could add attributes in own branch I could not spam until the custom schema management is ready.
